src/Idris/Parser.idr

```diff
@@ -1390,13 +1390,19 @@ topDecl fname indents
 -- collectDefs : List PDecl -> List PDecl
 collectDefs [] = []
 collectDefs (PDef annot cs :: ds)
-    = let (cs', rest) = spanBy isClause ds in
-          PDef annot (cs ++ concat cs') :: assert_total (collectDefs rest)
+    = let (csWithFC, rest) = spanBy isClause ds 
+          cs' = cs ++ concat (map snd csWithFC)
+          annot' = foldr mergeFC annot (map fst csWithFC)
+      in
+          PDef annot' cs' :: assert_total (collectDefs rest)
   where
-    isClause : PDecl -> Maybe (List PClause)
+    isClause : PDecl -> Maybe (FC, List PClause)
     isClause (PDef annot cs)
-        = Just cs
+        = Just (annot, cs)
     isClause _ = Nothing
+    mergeFC : FC -> FC -> FC
```

Comment:
`mergeFC` should go `Core.FC`. We should probably define the more general:
`mergeFC : FC -> FC -> Maybe FC` that fails in case the two input `FC` are
incompatible (if they reference different file names or if one is empty but
not the other).

We can then use `fromMaybe EmptyFC (mergeFC x y)` here.